src/OlStyleParser.ts

```diff
@@ -545,6 +555,86 @@ export class OlStyleParser implements StyleParser {
     return olStyleFct;
   }
 
+  /**
+   * Checks if a feature matches given filter expression(s)
+   * @param feature ol.Feature
+   * @param filter Filter
+   * @return boolean true if feature matches filter expression
+   */
+  geoStylerFilterToOlParserFilter(feature: ol.Feature, filter: Filter): boolean {
+    const operatorMapping = {
+      '&&': true,
+      '||': true,
+      '!': true
+    };
+
+    let matchesFilter: boolean = true;
+    const operator: Operator = filter[0];
+    let isNestedFilter: boolean = false;
+    if (operatorMapping[operator]) {
+      isNestedFilter = true;
+    }
+    if (isNestedFilter) {
+      switch (filter[0]) {
+        case '&&':
+          let intermediate = true;
+          let restFilter = filter.slice(1);
+          restFilter.forEach((f: Filter) => {
+            if (!this.geoStylerFilterToOlParserFilter(feature, f)) {
+              intermediate = false;
+            }
+          });
+          matchesFilter = intermediate;
+          break;
+        case '||':
+          intermediate = false;
+          restFilter = filter.slice(1);
+          restFilter.forEach((f: Filter) => {
+            if (this.geoStylerFilterToOlParserFilter(feature, f)) {
+              intermediate = true;
+            }
+          });
+          matchesFilter = intermediate;
+          break;
+        case '!':
+          matchesFilter = !this.geoStylerFilterToOlParserFilter(feature, filter[1]);
+          break;
+        default:
+          break;
+      }
+    } else {
+      const prop: any = feature.get(filter[1]);
+      switch (filter[0]) {
+        case '==':
+          matchesFilter = prop === filter[2];
+          break;
+        case '*=':
+          if (typeof filter[2] === 'string') {
+            matchesFilter = prop.includes(filter[2]);
```

Comment:
includs might be problematic, please check ancient browsers

Reply:
Replaced `includes` with the described polyfill from [mdn web docs](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/String/includes#Polyfill). Now it only uses indexOf, which is compatible with all not too old browsers.